### DISCUSSION TITLE
Error running Kyverno KUTTL tests: 'retrieving API resource for kyverno.io/v1, Kind=ClusterPolicy failed'

### DISCUSSION THREAD

**Discussion by J0SAL** [kyverno/policies#619] [Q&A]
I am trying to test the Kyverno policy mentioned in #546. 
However on running `kubectl kuttl test ./tests/e2e/` commands I'm getting the following error,

```
2023/05/21 21:26:59 running without a 'kuttl-test.yaml' configuration
2023/05/21 21:26:59 kutt-test config testdirs is overridden with args: [ ./tests/e2e/ ]
=== RUN   kuttl
    harness.go:494: starting setup
    harness.go:254: running tests with KIND.
    harness.go:180: temp folder created <path>\Temp\kuttl2034729971
    harness.go:162: Starting KIND cluster
    kind.go:66: Adding Containers to KIND...
    harness.go:294: Successful connection to cluster at: https://127.0.0.1:54304
    harness.go:375: running tests
=== RUN   kuttl/harness
=== NAME  kuttl
    harness.go:404: testsuite: . has 1 tests
=== RUN   kuttl/harness/lfx-test
=== PAUSE kuttl/harness/lfx-test
=== CONT  kuttl/harness/lfx-test
    logger.go:42: 21:27:46 | lfx-test | Creating namespace: kuttl-test-robust-boar
    logger.go:42: 21:27:46 | lfx-test/0-policy | starting test step 0-policy
    case.go:367: failed in step 0-policy
    case.go:369: retrieving API resource for kyverno.io/v1, Kind=ClusterPolicy failed: the server could not find the requested resource
    logger.go:42: 21:27:46 | lfx-test | lfx-test events from ns kuttl-test-robust-boar:
    logger.go:42: 21:27:46 | lfx-test | Deleting namespace: kuttl-test-robust-boar
=== NAME  kuttl
    harness.go:437: run tests finished
    harness.go:545: cleaning up
    harness.go:554: collecting cluster logs to kind-logs-1684684671
    harness.go:602: removing temp folder: "<path>\\Local\\Temp\\kuttl2034729971"
    harness.go:608: tearing down kind cluster
--- FAIL: kuttl (55.20s)
    --- FAIL: kuttl/harness (0.00s)
        --- FAIL: kuttl/harness/lfx-test (5.21s)
FAIL
Error: exit status 1

```

the code files I'm running are in `./tests/e2e/lfx-tests/` directory

Additional Information:
- Installed Kyverno using `kubectl create -f https://github.com/kyverno/kyverno/releases/download/v1.9.0/install.yaml` and running the command `kubectl get pods -n kyverno` gives following output 
```
NAME                                          READY   STATUS    RESTARTS   AGE
kyverno-6f444ccfc9-gfbp5                      1/1     Running   0          92m
kyverno-cleanup-controller-84798df489-fp8mt   1/1     Running   0          92m 
```
- Running `kubectl get clusterpolicy -n kyverno` shows the expected ClusterPolicy:
```
NAME             BACKGROUND   VALIDATE ACTION   READY   AGE
lfx-add-labels   true         Audit             true    93m
```
- Running `kubectl describe clusterpolicy lfx-add-labels -n kyverno` provides detailed information about the ClusterPolicy, including the spec and status.

Based on this information, I'm trying to understand the cause of the error. Please help me with Any insights or suggestions on resolving this issue. 

Thank you!

**Comment by chipzoller** [comment]
Failed in first step. What does that file look like?

**Reply by J0SAL** [reply]
here are my files, [github](https://github.com/J0SAL/lfx/tree/main/tests/e2e/lfx-test)

**Reply by chipzoller** [reply]
Is Kyverno installed when running the test?

**Reply by J0SAL** [reply]
Yes, I installed Kyverno using [YAMLs](https://kyverno.io/docs/installation/#install-kyverno-using-yamls).
Can this cause the problem?

**Reply by chipzoller** [reply]
No, it cannot. I just ran your test locally and so it does work. I suspect you have an issue with your cluster. Running kuttl already assumes 1) you have a working K8s cluster with the context set properly and 2) Kyverno installed.
